jo-rdan/Free-Mentor
# #168360958 create signup endpoint with persistence

#### What does this PR do?
It creates an endpoint for signing up with persistence to the database
#### Description of Task to be completed?
- create signup endpoint with persistence to database
- write unit tests for signup endpoint to test if the endpoint works as expected
#### How should this be manually tested?
- Clone this repository to your local computer
- Open the folder you cloned and start the server by typing **npm start**
- start **POSTMAN** and use this endpoint **/api/v2/auth/signup** and enter all fields correctly and it should return a **201** HTTP status code, otherwise it will respond with a relevant error message 
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
[#168360958](https://www.pivotaltracker.com/story/show/168360958)
### Screenshots (if appropriate)
**When everything is fine**
![image](https://user-images.githubusercontent.com/51836855/64545739-25303600-d2de-11e9-8c3e-959751804cc0.png)


**When error**
![image](https://user-images.githubusercontent.com/51836855/64545668-05007700-d2de-11e9-8d68-5849b087bce4.png)




#### Questions:
- N/A
